This research is about housing market of Canada. The reason why I come up this idea is that some many friends told me that the housing market is increasing a lot this year. Hence, I decide to look into the housing market and hopefully I can get some insight from the data.

Before analyzing data, I have to have some data on hand. So I feel like that I can try some web scraping from website which is posting housing information. The first website I tried is a website called 'Point2homes'. As a resident who have stayed in Nova Scotia for six years, I think that I should start this province first.

Here we go, let's try to get some data from it.

In [2]:
# Import requests
import requests
from bs4 import BeautifulSoup
import pandas as pd

This is a website page which includes 100 inforamtion of houses only in Nova Scotia.

In [163]:
url="https://www.point2homes.com/CA/Real-Estate-Maps.html?location=Nova%20Scotia&PropertyType=House&search_mode=location&SelectedView=map_view&LocationGeoId=205403&ajax=1"

Checking whether I can get information from it.

In [164]:
requests.get(url).status_code

200

Storing the data

In [165]:
data=requests.get(url).text

In [166]:
soup = BeautifulSoup(data, 'html.parser')

In [7]:
soup


<!DOCTYPE html>

<html lang="en">
<head>
<title>Nova Scotia Real Estate, Nova Scotia House for Sale by Point2</title>
<link href="https://www.point2homes.com/CA/Real-Estate-Maps.html" hreflang="en" rel="alternate"/>
<link href="https://www.point2homes.com/ES/CA/Real-Estate-Maps.html" hreflang="es" rel="alternate"/>
<!--Google One Tap-->
<script async="" defer="" src="https://accounts.google.com/gsi/client"></script>
<script src="https://apis.google.com/js/api:client.js"></script>
<!-- Analytics tracking code with Optimize plugin -->
<script>
    (function (i, s, o, g, r, a, m) {
        i['GoogleAnalyticsObject'] = r; i[r] = i[r] || function () {
            (i[r].q = i[r].q || []).push(arguments)
        }, i[r].l = 1 * new Date(); a = s.createElement(o),
            m = s.getElementsByTagName(o)[0]; a.async = 0; a.src = g; m.parentNode.insertBefore(a, m)
    })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');

    ga('create', 'UA-32199118-1', 'aut

After a little bit research, there are some information I found under the class called "item-details".

So let's store in the list called "tables"

In [167]:
tables= soup.find_all("div", class_="item-details")

Take a look at one information I have collected

In [168]:
tables[0]

<div class="item-details">
<div class="item-header-cnt">
<div class="characteristics-cnt">
<ul>
<li class="ic-beds" data-label="Beds">
<strong>2</strong> <span class="gray normal-lbl">Beds</span><span class="gray short-lbl">Bds</span>
</li>
<li class="ic-baths" data-label="Bath">
<strong>1</strong> <span class="gray normal-lbl">Bath</span><span class="gray short-lbl">Ba</span>
</li>
<li class="ic-sqft" data-label="Sqft">
<strong>1,200</strong> <span class="gray normal-lbl">Sqft</span><span class="gray short-lbl">Sqft</span>
</li>
<li class="ic-lotsize" data-label="Lot Size">
<strong>0.869 ac</strong> <span class="gray">Lot Size</span>
</li>
<li class="property-type ic-proptype">
            Residential
        </li>
</ul>
</div>
<div class="item-header-left-cnt">
<div class="price" data-price="$240,000 CAD">
<span class="green">
<span>
                $240,000 CAD
            </span>
</span>
</div>
<div class="item-address" data-address="2782 Hwy 362 " data-tracking="" data-url="/CA/Ho

In the beginning I found that this web page has 100 information of houses, so I want to check whether I got them all.

In [162]:
len(tables)

21

Now I want to get all the pages which are 2121 results of housing information. It should be 21 pages of 100 information plus 21 information at the 22nd page.

In [17]:
for i in range(1,23):
    url='https://www.point2homes.com/CA/Real-Estate-Maps.html?location=Nova+Scotia&PropertyType=House&search_mode=location&page=' + str(i) + '&SelectedView=map_view&LocationGeoId=205403&viewport=-67.2614835734375%2C42.61290244283285%2C-58.9228605265625%2C47.12543080145628&zoom=7&location_changed=&ajax=1'
    print(url)

https://www.point2homes.com/CA/Real-Estate-Maps.html?location=Nova+Scotia&PropertyType=House&search_mode=location&page=1&SelectedView=map_view&LocationGeoId=205403&viewport=-67.2614835734375%2C42.61290244283285%2C-58.9228605265625%2C47.12543080145628&zoom=7&location_changed=&ajax=1
https://www.point2homes.com/CA/Real-Estate-Maps.html?location=Nova+Scotia&PropertyType=House&search_mode=location&page=2&SelectedView=map_view&LocationGeoId=205403&viewport=-67.2614835734375%2C42.61290244283285%2C-58.9228605265625%2C47.12543080145628&zoom=7&location_changed=&ajax=1
https://www.point2homes.com/CA/Real-Estate-Maps.html?location=Nova+Scotia&PropertyType=House&search_mode=location&page=3&SelectedView=map_view&LocationGeoId=205403&viewport=-67.2614835734375%2C42.61290244283285%2C-58.9228605265625%2C47.12543080145628&zoom=7&location_changed=&ajax=1
https://www.point2homes.com/CA/Real-Estate-Maps.html?location=Nova+Scotia&PropertyType=House&search_mode=location&page=4&SelectedView=map_view&Location

The informaton can be collected are "ic-beds", "ic-baths", "ic-sqft", "ic-lotsize", "property-type ic-proptype", "price" and "item-address"

Let's store them in the data set

In [12]:
data = {}
soups = {}
tables = []
# Create a df to store info
Point2_data = pd.DataFrame(columns=['Beds','Bath','Sqft','Lot_Size','Type','Price','Address'])
for i in range(1,23):
    url='https://www.point2homes.com/CA/Real-Estate-Maps.html?location=Nova+Scotia&PropertyType=House&search_mode=location&page=' + str(i) + '&SelectedView=map_view&LocationGeoId=205403&viewport=-67.2614835734375%2C42.61290244283285%2C-58.9228605265625%2C47.12543080145628&zoom=7&location_changed=&ajax=1'
    data = requests.get(url).text
    soup = BeautifulSoup(data, 'html.parser')
    tables.append(soup.find_all("div", class_="item-details"))
    
    # extract information from each table
    for table in (tables[(i-1)]):
        if (len(str(table.find_all("li", class_='ic-beds')).split())<3):
            Nbed = "NaN"
        else:
            Nbed = str(table.find_all("li", class_='ic-beds')).split()[3][8:-9]
        
        if (len(str(table.find_all("li", class_='ic-baths')).split())<3):
            NBath = "NaN"
        else:
            NBath = str(table.find_all("li", class_='ic-baths')).split()[3][8:-9]
        
        if (len(str(table.find_all("li", class_='ic-sqft')).split())<3):
            NSqft = "NaN"
        else:
            NSqft = str(table.find_all("li", class_='ic-sqft')).split()[3][8:-9]
    
        if (len(str(table.find_all("li", class_='ic-lotsize')).split())<4):
            NLot_Size = "NaN"
        else:
            NLot_Size = str(table.find_all("li", class_='ic-lotsize')).split()[4][-4:]
        
        if (len(str(table.find_all("li", class_='ic-proptype')).split())<3):
            NType = "NaN"
        else:
            NType = str(table.find_all("li", class_='ic-proptype')).split()[3]


        NPrice = table.find("div", class_='price')['data-price']
        NAddress = table.find("div", class_='item-address')['data-address']
    # append data into the table 
        Point2_data = Point2_data.append({'Beds':Nbed,'Bath':NBath,'Sqft':NSqft,'Lot_Size':NLot_Size,'Type':NType,'Price':NPrice,'Address':NAddress},ignore_index=True)

In [13]:
Point2_data

Beds Bath   Sqft Lot_Size         Type           Price  \
0       2    1  1,200     .869  Residential    $240,000 CAD   
1       3    1  1,630     3.81  Residential    $225,000 CAD   
2       5    2  2,721     >107  Residential    $690,000 CAD   
3      15    6  6,630     .709  Residential    $495,000 CAD   
4       5    5  2,700     0.19  Residential    $499,950 CAD   
...   ...  ...    ...      ...          ...             ...   
2116  NaN  NaN    NaN     .803  Residential    $269,900 CAD   
2117    2    2  1,152      NaN  Residential    $179,900 CAD   
2118    3    1  1,470     0.15  Residential    $149,000 CAD   
2119    3    1  2,412      NaN       Single    $349,000 CAD   
2120    3    2  1,347      NaN  Residential  $1,200,000 CAD   

                       Address  
0                2782 Hwy 362   
1            1368 Forties Rd.   
2          2591 Clarence Road   
3     4616 Highway 1 Weymouth   
4            372 ST George St   
...                        ...  
2116               6666 Hwy 3   
2117           21 Bonnie Brae   
2118   186 Henry Hensey Drive   
2119            62 Smith Road   
2120    408 Cole Harbour Road   

[2121 rows x 7 columns]

Now I have the data, so we need to clean it.

Column "Sqft", 'Lot_Size', and 'Price' need to be reformated

In [14]:
for i in range(2121):
    Point2_data['Sqft'][i] = Point2_data['Sqft'][i].replace(',','')

In [15]:
for i in range(2121):
    Point2_data['Lot_Size'][i] = Point2_data['Lot_Size'][i].replace(',','')
    Point2_data['Lot_Size'][i] = Point2_data['Lot_Size'][i].replace('>','')

In [16]:
for i in range(2121):
    Point2_data['Price'][i] = Point2_data['Price'][i].replace(',','')
    Point2_data['Price'][i] = Point2_data['Price'][i].replace('$','')
    Point2_data['Price'][i] = Point2_data['Price'][i].replace('CAD','')

In [17]:
Point2_data

Beds Bath  Sqft Lot_Size         Type     Price                   Address
0       2    1  1200     .869  Residential   240000              2782 Hwy 362 
1       3    1  1630     3.81  Residential   225000          1368 Forties Rd. 
2       5    2  2721      107  Residential   690000        2591 Clarence Road 
3      15    6  6630     .709  Residential   495000   4616 Highway 1 Weymouth 
4       5    5  2700     0.19  Residential   499950          372 ST George St 
...   ...  ...   ...      ...          ...       ...                       ...
2116  NaN  NaN   NaN     .803  Residential   269900                6666 Hwy 3 
2117    2    2  1152      NaN  Residential   179900            21 Bonnie Brae 
2118    3    1  1470     0.15  Residential   149000    186 Henry Hensey Drive 
2119    3    1  2412      NaN       Single   349000             62 Smith Road 
2120    3    2  1347      NaN  Residential  1200000     408 Cole Harbour Road 

[2121 rows x 7 columns]

All the information is stored as string, so for some of columns, I need to convert to float.

Also, I need to find index of information need to be edited.

In [20]:
edited_index={}
listBed=[]
listBath=[]
listSqft=[]
listLot_Size=[]
listPrice=[]
for i in range(2121):
    # Beds column
    try:
        Point2_data["Beds"][i]=float(Point2_data["Beds"][i])
    except(ValueError):
        listBed.append(i)
        edited_index.update({'Bed': listBed})
    pass
    # Bath column
    try:
        Point2_data['Bath'][i]=float(Point2_data['Bath'][i])
    except(ValueError):
        listBath.append(i)
        edited_index.update({'Bath': listBath})
    pass
    # Sqft column
    try:
        Point2_data['Sqft'][i]=float(Point2_data['Sqft'][i])
    except(ValueError):
        listSqft.append(i)
        edited_index.update({'Sqft': listSqft})
    pass
    # Lot_Size column
    try:
        Point2_data['Lot_Size'][i]=float(Point2_data['Lot_Size'][i])
    except(ValueError):
        listLot_Size.append(i)
        edited_index.update({'Lot_Size': listLot_Size})
    pass
    # Price column
    try:
        Point2_data['Price'][i]=float(Point2_data['Price'][i])
    except(ValueError):
        listLot_Size.append(i)
        edited_index.update({'Price': listLot_Size})
    pass

Let's take a look what information I need to edit

In [21]:
for key in edited_index:
    print(key)

Lot_Size
Bath


Only 'Bath' and 'Lot_Size' need to be edited

In [22]:
print(len(edited_index['Bath']))
print(len(edited_index['Lot_Size']))

17
53


In [23]:
Point2_data.loc[edited_index['Lot_Size'],]

Beds  Bath    Sqft Lot_Size         Type      Price  \
8     4.0   2.0  3800.0      g17  Residential   849000.0   
52    4.0   2.0  2649.0      g18  Residential   799000.0   
55    3.0   2.0  3482.0      ng2  Residential  1250000.0   
58    2.0   1.0   948.0      ng5  Residential   168000.0   
94    4.0   2.0  1708.0      ng1  Residential   299000.0   
104   NaN   NaN     NaN      ng1  Residential    69000.0   
111   NaN   NaN     NaN      g16  Residential    59900.0   
116   NaN   NaN     NaN      ng5  Residential    54900.0   
148   3.0   4.0  4900.0      ng6  Residential  1485000.0   
155   4.0   4.0  2048.0      ng2  Residential   750000.0   
180   5.0   1.0  2190.0      g26  Residential   349900.0   
182   NaN   NaN  2378.0      g62  Residential   499000.0   
200   3.0   2.0  2412.0      ng8  Residential   699000.0   
219   3.0   2.0  1280.0      ng1  Residential   269000.0   
290   4.0   1.0  1800.0      g12  Residential   224900.0   
294   3.0   2.0  1590.0      ng3  Residential   599900.0   
298   NaN   NaN     NaN      g43  Residential   275000.0   
306   3.0   2.0  1512.0      ng4  Residential   375000.0   
331   3.0   1.0  1848.0      ng8  Residential   299000.0   
334   2.0   1.0   672.0      ng3  Residential   330000.0   
336   6.0   1.0  2600.0      g14  Residential   495000.0   
385   2.0   2.0  1200.0      g16  Residential   450000.0   
428   5.0   6.0  4642.0      g36  Residential  1475000.0   
694   4.0   4.0  2288.0      g90  Residential  1190000.0   
709   NaN   NaN  1299.0      ng1  Residential    69000.0   
771   2.0   1.0   896.0      ng1  Residential   329000.0   
872   3.0   1.0  1285.0      ng1  Residential   105000.0   
979   4.0   3.0  3106.0      ng1  Residential  1100000.0   
1043  7.0  10.0  2100.0      g37  Residential   850000.0   
1092  8.0   7.0  3208.0      ng2  Residential   369000.0   
1096  4.0   1.0  1290.0      ng2  Residential   159999.0   
1104  7.0   7.0  4032.0      ng5  Residential   980000.0   
1254  5.0   1.0  1846.0      ng1  Residential   255000.0   
1344  3.0   4.0  2680.0      g29  Residential   685000.0   
1385  4.0   3.0  2100.0      g16         Farm   575000.0   
1411  3.0   1.0  1848.0      ng8  Residential   299000.0   
1422  2.0   1.0  1120.0      g40  Residential   199000.0   
1487  3.0   2.0  1465.0      g66  Residential   849900.0   
1504  4.0   3.0  3860.0      g17  Residential   849000.0   
1580  3.0   1.0  1134.0      ng2  Residential   220000.0   
1588  3.0   2.0  1268.0      g23  Residential   448900.0   
1624  4.0   2.0  2600.0      ng4  Residential   449000.0   
1661  3.0   3.0  2464.0      ng3  Residential   824900.0   
1694  3.0   2.0  2856.0      g52  Residential   899000.0   
1887  3.0   2.0  1774.0      g43  Residential   697000.0   
1895  4.0   3.0  3021.0      g14  Residential  1375000.0   
1901  4.0   1.0  1290.0      ng2  Residential   395000.0   
2036  3.0   2.0  2856.0      g52  Residential   899000.0   
2067  3.0   1.0  2385.0      ng5  Residential   399000.0   
2072  3.0   1.0  1320.0      ng1  Residential   330000.0   
2093  4.0   1.0  1542.0      g10  Residential   395000.0   
2109  NaN   2.0  2144.0      ng4  Residential   244900.0   
2114  2.0   1.0   352.0      g14  Residential   298000.0   

                                             Address  
8                                    58 Pipers Lane   
52                                 3928 201 Highway   
55                                   6 Narrows Lane   
58                                 3585 Malagash Rd   
94                                    10 Grant Road   
104           62 Lake Doucette  Road, Lake Doucette   
111                          HWY 8 & Fisher Lake Rd   
116                         Lot 22 Fisher Lake Road   
148                                1852 Gospel Road   
155                               94 Clearland Road   
180                           6639 Upper Clyde Road   
182                               474 Harbour Acres   
200                              1396 Merland  Ro

In [24]:
tables[1][16]

<div class="item-details">
<div class="item-header-cnt">
<div class="characteristics-cnt">
<ul>
<li class="ic-lotsize" data-label="Lot Size">
<strong>5 ac</strong> <span class="gray">Lot Size</span>
</li>
<li class="property-type ic-proptype">
            Residential
        </li>
</ul>
</div>
<div class="item-header-left-cnt">
<div class="price" data-price="$54,900 CAD">
<span class="green">
<span>
                $54,900 CAD
            </span>
</span>
</div>
<div class="item-address" data-address="Lot 22 Fisher Lake Road " data-tracking="" data-url="/CA/Home-For-Sale/NS/Annapolis-Valley/Lot-22-Fisher-Lake-Road/102863083.html">
<h2><div class="address-container">Lot 22 Fisher Lake Road </div></h2>
</div>
</div>
</div>
</div>

In [336]:
tables[10][43]

<div class="item-details">
<div class="item-header-cnt">
<div class="characteristics-cnt">
<ul>
<li class="ic-beds" data-label="Beds">
<strong>7</strong> <span class="gray normal-lbl">Beds</span><span class="gray short-lbl">Bds</span>
</li>
<li class="ic-baths" data-label="Baths">
<strong>10</strong> <span class="gray normal-lbl">Baths</span><span class="gray short-lbl">Ba</span>
</li>
<li class="ic-sqft" data-label="Sqft">
<strong>2,100</strong> <span class="gray normal-lbl">Sqft</span><span class="gray short-lbl">Sqft</span>
</li>
<li class="ic-lotsize" data-label="Lot Size">
<strong>37 ac</strong> <span class="gray">Lot Size</span>
</li>
<li class="property-type ic-proptype">
            Residential
        </li>
</ul>
</div>
<div class="item-header-left-cnt">
<div class="price" data-price="$850,000 CAD">
<span class="green">
<span>
                $850,000 CAD
            </span>
</span>
</div>
<div class="item-address" data-address="45478 CABOT TRAIL Road ST. ANN'S " data-tracking

It seems like that Lot-size has "g", "n", or "ng" when collecting the data, so we need to remove them

In [25]:
for i in edited_index['Lot_Size']:
    Point2_data['Lot_Size'][i] = Point2_data['Lot_Size'][i].replace('g','')
    Point2_data['Lot_Size'][i] = Point2_data['Lot_Size'][i].replace('n','')
    Point2_data['Lot_Size'][i] = Point2_data['Lot_Size'][i].replace('ng','')

In [26]:
Point2_data.loc[edited_index['Lot_Size'],]

Beds  Bath    Sqft Lot_Size         Type      Price  \
8     4.0   2.0  3800.0       17  Residential   849000.0   
52    4.0   2.0  2649.0       18  Residential   799000.0   
55    3.0   2.0  3482.0        2  Residential  1250000.0   
58    2.0   1.0   948.0        5  Residential   168000.0   
94    4.0   2.0  1708.0        1  Residential   299000.0   
104   NaN   NaN     NaN        1  Residential    69000.0   
111   NaN   NaN     NaN       16  Residential    59900.0   
116   NaN   NaN     NaN        5  Residential    54900.0   
148   3.0   4.0  4900.0        6  Residential  1485000.0   
155   4.0   4.0  2048.0        2  Residential   750000.0   
180   5.0   1.0  2190.0       26  Residential   349900.0   
182   NaN   NaN  2378.0       62  Residential   499000.0   
200   3.0   2.0  2412.0        8  Residential   699000.0   
219   3.0   2.0  1280.0        1  Residential   269000.0   
290   4.0   1.0  1800.0       12  Residential   224900.0   
294   3.0   2.0  1590.0        3  Residential   599900.0   
298   NaN   NaN     NaN       43  Residential   275000.0   
306   3.0   2.0  1512.0        4  Residential   375000.0   
331   3.0   1.0  1848.0        8  Residential   299000.0   
334   2.0   1.0   672.0        3  Residential   330000.0   
336   6.0   1.0  2600.0       14  Residential   495000.0   
385   2.0   2.0  1200.0       16  Residential   450000.0   
428   5.0   6.0  4642.0       36  Residential  1475000.0   
694   4.0   4.0  2288.0       90  Residential  1190000.0   
709   NaN   NaN  1299.0        1  Residential    69000.0   
771   2.0   1.0   896.0        1  Residential   329000.0   
872   3.0   1.0  1285.0        1  Residential   105000.0   
979   4.0   3.0  3106.0        1  Residential  1100000.0   
1043  7.0  10.0  2100.0       37  Residential   850000.0   
1092  8.0   7.0  3208.0        2  Residential   369000.0   
1096  4.0   1.0  1290.0        2  Residential   159999.0   
1104  7.0   7.0  4032.0        5  Residential   980000.0   
1254  5.0   1.0  1846.0        1  Residential   255000.0   
1344  3.0   4.0  2680.0       29  Residential   685000.0   
1385  4.0   3.0  2100.0       16         Farm   575000.0   
1411  3.0   1.0  1848.0        8  Residential   299000.0   
1422  2.0   1.0  1120.0       40  Residential   199000.0   
1487  3.0   2.0  1465.0       66  Residential   849900.0   
1504  4.0   3.0  3860.0       17  Residential   849000.0   
1580  3.0   1.0  1134.0        2  Residential   220000.0   
1588  3.0   2.0  1268.0       23  Residential   448900.0   
1624  4.0   2.0  2600.0        4  Residential   449000.0   
1661  3.0   3.0  2464.0        3  Residential   824900.0   
1694  3.0   2.0  2856.0       52  Residential   899000.0   
1887  3.0   2.0  1774.0       43  Residential   697000.0   
1895  4.0   3.0  3021.0       14  Residential  1375000.0   
1901  4.0   1.0  1290.0        2  Residential   395000.0   
2036  3.0   2.0  2856.0       52  Residential   899000.0   
2067  3.0   1.0  2385.0        5  Residential   399000.0   
2072  3.0   1.0  1320.0        1  Residential   330000.0   
2093  4.0   1.0  1542.0       10  Residential   395000.0   
2109  NaN   2.0  2144.0        4  Residential   244900.0   
2114  2.0   1.0   352.0       14  Residential   298000.0   

                                             Address  
8                                    58 Pipers Lane   
52                                 3928 201 Highway   
55                                   6 Narrows Lane   
58                                 3585 Malagash Rd   
94                                    10 Grant Road   
104           62 Lake Doucette  Road, Lake Doucette   
111                          HWY 8 & Fisher Lake Rd   
116                         Lot 22 Fisher Lake Road   
148                                1852 Gospel Road   
155                               94 Clearland Road   
180                           6639 Upper Clyde Road   
182                               474 Harbour Acres   
200                              1396 Merland  Ro

Let's convert them to float

In [27]:
for i in edited_index['Lot_Size']:
    Point2_data['Lot_Size'][i]=float(Point2_data['Lot_Size'][i])

In [28]:
Point2_data.loc[edited_index['Bath'],]

Beds Bath Sqft Lot_Size         Type     Price  \
70    NaN   el  NaN      NaN  Residential  745900.0   
83    6.0   el  NaN      1.8  Residential  495000.0   
390   2.0   el  NaN      NaN       Single  369900.0   
661   3.0   el  NaN      NaN  Residential  179900.0   
756   NaN    e  NaN      6.5       single  675000.0   
829   3.0    e  NaN      NaN  Residential  249000.0   
1130  3.0   el  NaN      NaN  Residential  378000.0   
1197  5.0   el  NaN      NaN  Residential  299000.0   
1200  5.0    e  NaN      NaN       Single   50000.0   
1491  5.0   el  NaN      NaN  Residential  249900.0   
1543  4.0    e  NaN      NaN  Residential  699000.0   
1556  3.0    e  NaN      NaN  Residential  199000.0   
1677  5.0    e  NaN      NaN  Residential  174900.0   
1793  3.0   el  NaN      NaN  Residential  249000.0   
1853  3.0    e  NaN      NaN  Residential  199900.0   
2105  2.0    e  NaN    0.895  Residential  249900.0   
2111  4.0   el  NaN    0.331  Residential  325000.0   

                                  Address  
70                    74 Parkdale Avenue   
83                             147 Hwy 8   
390                     10473 HIGHWAY 16   
661               11822 CABOT TRAIL Road   
756                   280 GREEN BAY Road   
829          96 Main Street, Guysborough   
1130                    2 Shieling Court   
1197         422 Highway 316 Loch Katrin   
1200                      5296 Highway 1   
1491                   130 Larry's River   
1543                  130 Kavanaugh Lane   
1556   36 Carr Road, St. Francis Harbour   
1677           8153 highway 7 sherbrooke   
1793                  14750 Highway 316    
1853  9 Ocean View Drive, Port Bickerton   
2105        199 Old Trunk Rd Elmsdale NS   
2111                         244 Chester

In [29]:
tables[0][70]

<div class="item-details">
<div class="item-header-cnt">
<div class="characteristics-cnt">
<ul>
<li class="ic-baths nosq" data-label="Baths">
<strong>3</strong> <span class="gray normal-lbl">Baths</span><span class="gray short-lbl">Ba</span>
</li>
<li class="property-type ic-proptype">
            Residential
        </li>
</ul>
</div>
<div class="item-header-left-cnt">
<div class="price" data-price="$745,900 CAD">
<span class="green">
<span>
                $745,900 CAD
            </span>
</span>
</div>
<div class="item-address" data-address="74 Parkdale Avenue " data-tracking="" data-url="/CA/Home-For-Sale/NS/Eastern-Shore/Greater-Halifax/74-Parkdale-Avenue/111827706.html">
<h2><div class="address-container">74 Parkdale Avenue </div></h2>
</div>
</div>
</div>
</div>

Let's store the correct value

In [30]:
str(tables[0][70].find_all("li", class_='ic-baths')).split()[4][8:-9]

'3'

In [31]:
for i in edited_index['Bath']:
    Point2_data['Bath'][i]=str(tables[int(i/100)][(i-100*int(i/100))].find_all("li", class_='ic-baths')).split()[4][8:-9]

In [32]:
Point2_data.loc[edited_index['Bath'],]

Beds Bath Sqft Lot_Size         Type     Price  \
70    NaN    3  NaN      NaN  Residential  745900.0   
83    6.0    5  NaN      1.8  Residential  495000.0   
390   2.0    3  NaN      NaN       Single  369900.0   
661   3.0    2  NaN      NaN  Residential  179900.0   
756   NaN    1  NaN      6.5       single  675000.0   
829   3.0    1  NaN      NaN  Residential  249000.0   
1130  3.0    2  NaN      NaN  Residential  378000.0   
1197  5.0    3  NaN      NaN  Residential  299000.0   
1200  5.0    1  NaN      NaN       Single   50000.0   
1491  5.0    2  NaN      NaN  Residential  249900.0   
1543  4.0    1  NaN      NaN  Residential  699000.0   
1556  3.0    1  NaN      NaN  Residential  199000.0   
1677  5.0    1  NaN      NaN  Residential  174900.0   
1793  3.0    2  NaN      NaN  Residential  249000.0   
1853  3.0    1  NaN      NaN  Residential  199900.0   
2105  2.0    1  NaN    0.895  Residential  249900.0   
2111  4.0    2  NaN    0.331  Residential  325000.0   

                                  Address  
70                    74 Parkdale Avenue   
83                             147 Hwy 8   
390                     10473 HIGHWAY 16   
661               11822 CABOT TRAIL Road   
756                   280 GREEN BAY Road   
829          96 Main Street, Guysborough   
1130                    2 Shieling Court   
1197         422 Highway 316 Loch Katrin   
1200                      5296 Highway 1   
1491                   130 Larry's River   
1543                  130 Kavanaugh Lane   
1556   36 Carr Road, St. Francis Harbour   
1677           8153 highway 7 sherbrooke   
1793                  14750 Highway 316    
1853  9 Ocean View Drive, Port Bickerton   
2105        199 Old Trunk Rd Elmsdale NS   
2111                         244 Chester

Let's take a look how many "NaN" values

In [33]:
for column in Point2_data:
    print(str(column) + ' column has '+ str(Point2_data[column].isnull().sum()) + ' NaNs')

Beds column has 37 NaNs
Bath column has 34 NaNs
Sqft column has 34 NaNs
Lot_Size column has 1850 NaNs
Type column has 0 NaNs
Price column has 0 NaNs
Address column has 0 NaNs


I think I don't need 'Lot_Size' and 'Type' for further analysis 

In [34]:
Point2_data = Point2_data.drop(['Lot_Size','Type'],axis=1)

In [35]:
for column in Point2_data:
    print(str(column) + ' column has '+ str(Point2_data[column].isnull().sum()) + ' NaNs')

Beds column has 37 NaNs
Bath column has 34 NaNs
Sqft column has 34 NaNs
Price column has 0 NaNs
Address column has 0 NaNs


Let's take a look at the NaN columns

In [49]:
Point2_data.loc[Point2_data['Beds'].isnull(),'value_is_NaN'] = 'Yes'
Point2_data.loc[Point2_data['Bath'].isnull(),'value_is_NaN'] = 'Yes'
Point2_data.loc[Point2_data['Sqft'].isnull(),'value_is_NaN'] = 'Yes'

In [50]:
Point2_data

Beds Bath    Sqft      Price                   Address value_is_NaN
0      2.0  1.0  1200.0   240000.0             2782 Hwy 362           NaN
1      3.0  1.0  1630.0   225000.0         1368 Forties Rd.           NaN
2      5.0  2.0  2721.0   690000.0       2591 Clarence Road           NaN
3     15.0  6.0  6630.0   495000.0  4616 Highway 1 Weymouth           NaN
4      5.0  5.0  2700.0   499950.0         372 ST George St           NaN
...    ...  ...     ...        ...                       ...          ...
2116   NaN  NaN     NaN   269900.0               6666 Hwy 3           Yes
2117   2.0  2.0  1152.0   179900.0           21 Bonnie Brae           NaN
2118   3.0  1.0  1470.0   149000.0   186 Henry Hensey Drive           NaN
2119   3.0  1.0  2412.0   349000.0            62 Smith Road           NaN
2120   3.0  2.0  1347.0  1200000.0    408 Cole Harbour Road           NaN

[2121 rows x 6 columns]

In [51]:
Row_Nan=[]
for i in range(2121):
    if Point2_data['value_is_NaN'][i] == 'Yes':
        Row_Nan.append(i)

In [52]:
len(Row_Nan)

59

In [53]:
Point2_data.loc[Row_Nan,['Beds','Bath','Sqft']]

Beds Bath    Sqft
70    NaN    3     NaN
82    NaN  NaN     NaN
83    6.0    5     NaN
104   NaN  NaN     NaN
111   NaN  NaN     NaN
116   NaN  NaN     NaN
134   NaN  NaN     NaN
182   NaN  NaN  2378.0
244   NaN  NaN     NaN
298   NaN  NaN     NaN
302   NaN  NaN     NaN
315   NaN  NaN     NaN
329   NaN  NaN     NaN
337   NaN  NaN     NaN
390   2.0    3     NaN
404   3.0  NaN  1072.0
449   4.0  NaN   742.0
600   NaN  2.0  1950.0
661   3.0    2     NaN
709   NaN  NaN  1299.0
712   NaN  1.0  2200.0
741   NaN  1.0  1986.0
751   NaN  1.0   651.0
756   NaN    1     NaN
813   NaN  1.0   400.0
829   3.0    1     NaN
847   NaN  NaN  2374.0
985   3.0  NaN  1390.0
990   2.0  NaN     NaN
1008  NaN  NaN  1955.0
1122  NaN  NaN   320.0
1130  3.0    2     NaN
1185  NaN  NaN     NaN
1197  5.0    3     NaN
1200  5.0    1     NaN
1214  NaN  NaN  1202.0
1226  1.0  NaN   240.0
1287  NaN  NaN  2016.0
1402  5.0  NaN  1748.0
1491  5.0    2     NaN
1543  4.0    1     NaN
1556  3.0    1     NaN
1574  NaN  NaN  2128.0
1677  5.0    1     NaN
1776  NaN  NaN  3788.0
1793  3.0    2     NaN
1807  NaN  NaN  1004.0
1840  NaN  NaN  3400.0
1853  3.0    1     NaN
2013  NaN  1.0  1702.0
2026  NaN  NaN     NaN
2063  NaN  NaN     NaN
2064  NaN  NaN     NaN
2095  3.0  NaN  2125.0
2099  NaN  1.0  1578.0
2105  2.0    1     NaN
2109  NaN  2.0  2144.0
2111  4.0    2     NaN
2116  NaN  NaN     NaN

In [47]:
# a = Point2_data

In [55]:
Point2_data.dropna(subset=['Beds','Bath','Sqft'], inplace=True)

In [56]:
Point2_data

Beds Bath    Sqft      Price                   Address value_is_NaN
0      2.0  1.0  1200.0   240000.0             2782 Hwy 362           NaN
1      3.0  1.0  1630.0   225000.0         1368 Forties Rd.           NaN
2      5.0  2.0  2721.0   690000.0       2591 Clarence Road           NaN
3     15.0  6.0  6630.0   495000.0  4616 Highway 1 Weymouth           NaN
4      5.0  5.0  2700.0   499950.0         372 ST George St           NaN
...    ...  ...     ...        ...                       ...          ...
2115   3.0  2.0  1655.0   484900.0  Lot 19 242 Curtis Drive           NaN
2117   2.0  2.0  1152.0   179900.0           21 Bonnie Brae           NaN
2118   3.0  1.0  1470.0   149000.0   186 Henry Hensey Drive           NaN
2119   3.0  1.0  2412.0   349000.0            62 Smith Road           NaN
2120   3.0  2.0  1347.0  1200000.0    408 Cole Harbour Road           NaN

[2062 rows x 6 columns]

In [59]:
Point2_data.reset_index()

index  Beds Bath    Sqft      Price                   Address  \
0         0   2.0  1.0  1200.0   240000.0             2782 Hwy 362    
1         1   3.0  1.0  1630.0   225000.0         1368 Forties Rd.    
2         2   5.0  2.0  2721.0   690000.0       2591 Clarence Road    
3         3  15.0  6.0  6630.0   495000.0  4616 Highway 1 Weymouth    
4         4   5.0  5.0  2700.0   499950.0         372 ST George St    
...     ...   ...  ...     ...        ...                       ...   
2057   2115   3.0  2.0  1655.0   484900.0  Lot 19 242 Curtis Drive    
2058   2117   2.0  2.0  1152.0   179900.0           21 Bonnie Brae    
2059   2118   3.0  1.0  1470.0   149000.0   186 Henry Hensey Drive    
2060   2119   3.0  1.0  2412.0   349000.0            62 Smith Road    
2061   2120   3.0  2.0  1347.0  1200000.0    408 Cole Harbour Road    

     value_is_NaN  
0             NaN  
1             NaN  
2             NaN  
3             NaN  
4             NaN  
...           ...  
2057          NaN  
2058          NaN  
2059          NaN  
2060          NaN  
2061          NaN  

[2062 rows x 7 columns]

In [60]:
Point2_data.drop

Beds Bath    Sqft      Price                   Address value_is_NaN
0      2.0  1.0  1200.0   240000.0             2782 Hwy 362           NaN
1      3.0  1.0  1630.0   225000.0         1368 Forties Rd.           NaN
2      5.0  2.0  2721.0   690000.0       2591 Clarence Road           NaN
3     15.0  6.0  6630.0   495000.0  4616 Highway 1 Weymouth           NaN
4      5.0  5.0  2700.0   499950.0         372 ST George St           NaN
...    ...  ...     ...        ...                       ...          ...
2115   3.0  2.0  1655.0   484900.0  Lot 19 242 Curtis Drive           NaN
2117   2.0  2.0  1152.0   179900.0           21 Bonnie Brae           NaN
2118   3.0  1.0  1470.0   149000.0   186 Henry Hensey Drive           NaN
2119   3.0  1.0  2412.0   349000.0            62 Smith Road           NaN
2120   3.0  2.0  1347.0  1200000.0    408 Cole Harbour Road           NaN

[2062 rows x 6 columns]

In [62]:
Point2_data = Point2_data.drop(['value_is_NaN'],axis=1)

In [68]:
Point2_data= Point2_data.reset_index()

In [77]:
Point2_data.reset_index()

index  Beds Bath    Sqft      Price                   Address
0         0   2.0  1.0  1200.0   240000.0             2782 Hwy 362 
1         1   3.0  1.0  1630.0   225000.0         1368 Forties Rd. 
2         2   5.0  2.0  2721.0   690000.0       2591 Clarence Road 
3         3  15.0  6.0  6630.0   495000.0  4616 Highway 1 Weymouth 
4         4   5.0  5.0  2700.0   499950.0         372 ST George St 
...     ...   ...  ...     ...        ...                       ...
2057   2057   3.0  2.0  1655.0   484900.0  Lot 19 242 Curtis Drive 
2058   2058   2.0  2.0  1152.0   179900.0           21 Bonnie Brae 
2059   2059   3.0  1.0  1470.0   149000.0   186 Henry Hensey Drive 
2060   2060   3.0  1.0  2412.0   349000.0            62 Smith Road 
2061   2061   3.0  2.0  1347.0  1200000.0    408 Cole Harbour Road 

[2062 rows x 6 columns]

In [79]:
Point2_data.set_index

<bound method DataFrame.set_index of       Beds Bath    Sqft      Price                   Address
0      2.0  1.0  1200.0   240000.0             2782 Hwy 362 
1      3.0  1.0  1630.0   225000.0         1368 Forties Rd. 
2      5.0  2.0  2721.0   690000.0       2591 Clarence Road 
3     15.0  6.0  6630.0   495000.0  4616 Highway 1 Weymouth 
4      5.0  5.0  2700.0   499950.0         372 ST George St 
...    ...  ...     ...        ...                       ...
2057   3.0  2.0  1655.0   484900.0  Lot 19 242 Curtis Drive 
2058   2.0  2.0  1152.0   179900.0           21 Bonnie Brae 
2059   3.0  1.0  1470.0   149000.0   186 Henry Hensey Drive 
2060   3.0  1.0  2412.0   349000.0            62 Smith Road 
2061   3.0  2.0  1347.0  1200000.0    408 Cole Harbour Road 

[2062 rows x 5 columns]>

In [81]:
Point2_data.to_csv('Point2_data.csv')

In [93]:
Point2_data= pd.read_csv('/users/aeternus/Desktop/Project of NS housing/Point2_data.csv')

In [94]:
Point2_data.isnull().sum()

Unnamed: 0    0
Beds          0
Bath          0
Sqft          0
Price         0
Address       0
dtype: int64

In [97]:
Point2_data = Point2_data.drop(['Unnamed: 0'],axis=1)

In [98]:
Point2_data

Beds  Bath    Sqft      Price                   Address
0      2.0   1.0  1200.0   240000.0             2782 Hwy 362 
1      3.0   1.0  1630.0   225000.0         1368 Forties Rd. 
2      5.0   2.0  2721.0   690000.0       2591 Clarence Road 
3     15.0   6.0  6630.0   495000.0  4616 Highway 1 Weymouth 
4      5.0   5.0  2700.0   499950.0         372 ST George St 
...    ...   ...     ...        ...                       ...
2057   3.0   2.0  1655.0   484900.0  Lot 19 242 Curtis Drive 
2058   2.0   2.0  1152.0   179900.0           21 Bonnie Brae 
2059   3.0   1.0  1470.0   149000.0   186 Henry Hensey Drive 
2060   3.0   1.0  2412.0   349000.0            62 Smith Road 
2061   3.0   2.0  1347.0  1200000.0    408 Cole Harbour Road 

[2062 rows x 5 columns]

In [99]:
Point2_data.dtypes

Beds       float64
Bath       float64
Sqft       float64
Price      float64
Address     object
dtype: object

Now let's try to find the coordinates of address

In [111]:
from geopy.geocoders import Nominatim

In [112]:
location = geolocator.geocode("2591 Clarence Road")

In [163]:
Point2_data

Beds  Bath    Sqft      Price                   Address
0      2.0   1.0  1200.0   240000.0             2782 Hwy 362 
1      3.0   1.0  1630.0   225000.0         1368 Forties Rd. 
2      5.0   2.0  2721.0   690000.0       2591 Clarence Road 
3     15.0   6.0  6630.0   495000.0  4616 Highway 1 Weymouth 
4      5.0   5.0  2700.0   499950.0         372 ST George St 
...    ...   ...     ...        ...                       ...
2057   3.0   2.0  1655.0   484900.0  Lot 19 242 Curtis Drive 
2058   2.0   2.0  1152.0   179900.0           21 Bonnie Brae 
2059   3.0   1.0  1470.0   149000.0   186 Henry Hensey Drive 
2060   3.0   1.0  2412.0   349000.0            62 Smith Road 
2061   3.0   2.0  1347.0  1200000.0    408 Cole Harbour Road 

[2062 rows x 5 columns]

In [168]:
for i in range(2062):
    Point2_data['Address'][i]= str(Point2_data['Address'][i]) + ' Nova Scotia'

<ipython-input-168-7fc79547d301>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Point2_data['Address'][i]= str(Point2_data['Address'][i]) + ' Nova Scotia'


In [169]:
Point2_data

Beds  Bath    Sqft      Price                               Address
0      2.0   1.0  1200.0   240000.0             2782 Hwy 362  Nova Scotia
1      3.0   1.0  1630.0   225000.0         1368 Forties Rd.  Nova Scotia
2      5.0   2.0  2721.0   690000.0       2591 Clarence Road  Nova Scotia
3     15.0   6.0  6630.0   495000.0  4616 Highway 1 Weymouth  Nova Scotia
4      5.0   5.0  2700.0   499950.0         372 ST George St  Nova Scotia
...    ...   ...     ...        ...                                   ...
2057   3.0   2.0  1655.0   484900.0  Lot 19 242 Curtis Drive  Nova Scotia
2058   2.0   2.0  1152.0   179900.0           21 Bonnie Brae  Nova Scotia
2059   3.0   1.0  1470.0   149000.0   186 Henry Hensey Drive  Nova Scotia
2060   3.0   1.0  2412.0   349000.0            62 Smith Road  Nova Scotia
2061   3.0   2.0  1347.0  1200000.0    408 Cole Harbour Road  Nova Scotia

[2062 rows x 5 columns]

In [175]:
coordinates = []
for i in range(2062):
    try:
        location = geolocator.geocode(Point2_data['Address'][i])
        coordinates.append((location.latitude, location.longitude)) 
    except(AttributeError):
        coordinates.append(('NaN','NaN'))
    pass

In [196]:
for i in range(2062):
    Point2_data['latitude'][i] = coordinates[i][0]
    Point2_data['longitude'][i] = coordinates[i][1]

<ipython-input-196-3d2f60542131>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Point2_data['latitude'][i] = coordinates[i][0]
<ipython-input-196-3d2f60542131>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Point2_data['longitude'][i] = coordinates[i][1]


In [197]:
Point2_data

Beds  Bath    Sqft      Price                               Address  \
0      2.0   1.0  1200.0   240000.0             2782 Hwy 362  Nova Scotia   
1      3.0   1.0  1630.0   225000.0         1368 Forties Rd.  Nova Scotia   
2      5.0   2.0  2721.0   690000.0       2591 Clarence Road  Nova Scotia   
3     15.0   6.0  6630.0   495000.0  4616 Highway 1 Weymouth  Nova Scotia   
4      5.0   5.0  2700.0   499950.0         372 ST George St  Nova Scotia   
...    ...   ...     ...        ...                                   ...   
2057   3.0   2.0  1655.0   484900.0  Lot 19 242 Curtis Drive  Nova Scotia   
2058   2.0   2.0  1152.0   179900.0           21 Bonnie Brae  Nova Scotia   
2059   3.0   1.0  1470.0   149000.0   186 Henry Hensey Drive  Nova Scotia   
2060   3.0   1.0  2412.0   349000.0            62 Smith Road  Nova Scotia   
2061   3.0   2.0  1347.0  1200000.0    408 Cole Harbour Road  Nova Scotia   

       latitude  longitude  
0     45.046471 -65.064869  
1     44.734216 -64.539807  
2     44.904340 -65.218431  
3     44.411117 -65.995788  
4     44.742408 -65.516158  
...         ...        ...  
2057        NaN        NaN  
2058  44.648317 -63.534459  
2059  44.037933 -64.722590  
2060  44.440700 -64.892060  
2061  45.245968 -61.260294  

[2062 rows x 7 columns]

In [176]:
coordinates

[(45.04647068, -65.06486906),
 (44.73421590607276, -64.53980684391512),
 (44.904340325889684, -65.21843098914215),
 (44.41111729197513, -65.99578783018084),
 (44.74240775, -65.51615788104628),
 (44.701909799999996, -63.728475114773595),
 (44.56096552427477, -64.84347120234645),
 (43.926649117906614, -66.10820905431348),
 (46.23683339217818, -61.28021017295488),
 (44.6743945, -63.463930942300955),
 (44.583207359999996, -64.192734),
 (46.647378308, -60.98013928),
 (44.68546105, -63.66772673926155),
 (44.45253405206259, -65.81600430481588),
 (44.7834012, -63.701700387908495),
 (45.35677344, -63.300789359999996),
 (44.631506124458475, -65.76771370979637),
 (44.807210999999995, -63.75135237627681),
 (46.244764571428576, -60.279885),
 (44.59524772181067, -65.60412267736604),
 (44.6774049, -63.550831045202045),
 (45.683971675672844, -62.69904306068488),
 (45.3559408, -63.2821802),
 (45.648273, -64.051317),
 (44.6673423, -63.6150985),
 (44.832768, -63.60059222201069),
 ('NaN', 'NaN'),
 (44.798

In [224]:
CheckNan = Point2_data['latitude'].isnull()

In [228]:
checklist=[]
for i, check in enumerate(CheckNan):
    if check is True:
        checklist.append(i)

In [258]:
for i in range(2062):
    Point2_data['Address'][i] = Point2_data['Address'][i].replace('NS','NS, Canada')

<ipython-input-258-424abc5fd949>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Point2_data['Address'][i] = Point2_data['Address'][i].replace('NS','NS, Canada')


In [295]:
Point2_data.loc[checklist,:]

Beds  Bath    Sqft     Price  \
26     3.0   2.0  1372.0  209900.0   
47     2.0   2.0   804.0  309900.0   
50     6.0   3.0  3417.0  539850.0   
54     2.0   1.0   915.0  195000.0   
71     3.0   3.0  3129.0  980000.0   
...    ...   ...     ...       ...   
2039   3.0   1.0  1999.0  209900.0   
2043   3.0   3.0  1792.0  499900.0   
2050   2.0   1.0   832.0  334900.0   
2053   3.0   3.0  1792.0  499900.0   
2057   3.0   2.0  1655.0  484900.0   

                                                Address  latitude  longitude  
26                        4944 Medway River  NS, Canada       NaN        NaN  
47               12 - 56 Meek Arm Trail Way  NS, Canada       NaN        NaN  
50                      1086 Marie Crescent  NS, Canada       NaN        NaN  
54                          6523 16 Highway  NS, Canada       NaN        NaN  
71                         86 Freddies Lane  NS, Canada       NaN        NaN  
...                                                 ...       ...        ...  
2039  113 Kinsac Road, Beaverbank NS, Canada  NS, Ca...       NaN        NaN  
2043                Lot 26 205 Curtis Drive  NS, Canada       NaN        NaN  
2050                     12-66 Meek Arm Way  NS, Canada       NaN        NaN  
2053                Lot 29 187 Curtis Drive  NS, Canada       NaN        NaN  
2057                Lot 19 242 Curtis Drive  NS, Canada       NaN        NaN  

[230 rows x 7 columns]

Those 230 address cannot find on Google Map. 

## In other words, those locations need to update and make them readable on Google Map
## Or the Google Map need to includes those address.

In [312]:
Point2_data.to_csv('Point2_data.csv',index=False)

In [313]:
Point2_data= pd.read_csv('/users/aeternus/Desktop/Project of NS housing/Point2_data.csv')

In [314]:
Point2_data

Beds  Bath    Sqft      Price                              Address  \
0      2.0   1.0  1200.0   240000.0             2782 Hwy 362  NS, Canada   
1      3.0   1.0  1630.0   225000.0         1368 Forties Rd.  NS, Canada   
2      5.0   2.0  2721.0   690000.0       2591 Clarence Road  NS, Canada   
3     15.0   6.0  6630.0   495000.0  4616 Highway 1 Weymouth  NS, Canada   
4      5.0   5.0  2700.0   499950.0         372 ST George St  NS, Canada   
...    ...   ...     ...        ...                                  ...   
2057   3.0   2.0  1655.0   484900.0  Lot 19 242 Curtis Drive  NS, Canada   
2058   2.0   2.0  1152.0   179900.0           21 Bonnie Brae  NS, Canada   
2059   3.0   1.0  1470.0   149000.0   186 Henry Hensey Drive  NS, Canada   
2060   3.0   1.0  2412.0   349000.0            62 Smith Road  NS, Canada   
2061   3.0   2.0  1347.0  1200000.0    408 Cole Harbour Road  NS, Canada   

       latitude  longitude  
0     45.046471 -65.064869  
1     44.734216 -64.539807  
2     44.904340 -65.218431  
3     44.411117 -65.995788  
4     44.742408 -65.516158  
...         ...        ...  
2057        NaN        NaN  
2058  44.648317 -63.534459  
2059  44.037933 -64.722590  
2060  44.440700 -64.892060  
2061  45.245968 -61.260294  

[2062 rows x 7 columns]